keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ApplicationServiceTests.kt

```diff
@@ -123,250 +116,401 @@ class ApplicationServiceTests : JUnit5Minutests {
+          expectThat(summaries) {
+            first()
+              .withVersionInEnvironment(version4, "test") {
+                state.isEqualTo(PENDING.name.toLowerCase())
+              }
+              .withVersionInEnvironment(version4, "staging") {
+                state.isEqualTo(PENDING.name.toLowerCase())
+              }
+              .withVersionInEnvironment(version3, "staging") {
+                state.isEqualTo(PENDING.name.toLowerCase())
+              }
+              .withVersionInEnvironment(version4, "production") {
+                state.isEqualTo(PENDING.name.toLowerCase())
+              }
+              .withVersionInEnvironment(version3, "production") {
+                state.isEqualTo(PENDING.name.toLowerCase())
+              }
+              .withVersionInEnvironment(version2, "production") {
+                state.isEqualTo(PENDING.name.toLowerCase())
+              }
+          }
+        }
+
+        test("stateless constraint details are included in the summary") {
+          val summaries = applicationService.getArtifactSummariesFor(application)
+
+          expectThat(summaries.first { it.name == artifact.name })
+            .withVersionInEnvironment(version1, "production") {
+              get { statelessConstraints }
+                .hasSize(1)
+                .first().and {
+                  get { type }.isEqualTo("depends-on")
+                  get { currentlyPassing }.isTrue()
+                }
+            }
+            .withVersionInEnvironment(version2, "production") {
+              get { statelessConstraints }
+                .hasSize(1)
+                .first().and {
+                  get { type }.isEqualTo("depends-on")
+                  get { currentlyPassing }.isFalse()
+                }
+            }
+        }
+
+        test("stateful constraint details are included in the summary") {
+          val summaries = applicationService.getArtifactSummariesFor(application)
+
+          expectThat(summaries.first { it.name == artifact.name })
+            .withVersionInEnvironment(version1, "production") {
+              get { statefulConstraints }
+                .first { it.type == "pipeline" }
+                .get { status }.isEqualTo(PASS)
+            }
+            .withVersionInEnvironment(version2, "production") {
+              get { statefulConstraints }
+                .first { it.type == "pipeline" }
+                .get { status }.isEqualTo(PENDING)
+            }
+        }
+
+        test("non-evaluated stateful constraint details are included in the summary") {
+          val summaries = applicationService.getArtifactSummariesFor(application)
+
+          expectThat(summaries.first { it.name == artifact.name })
+            .withVersionInEnvironment(version1, "production") {
+              get { statefulConstraints }
+                .first { it.type == "manual-judgement" }
+                .get { status }.isEqualTo(NOT_EVALUATED)
+            }
         }
```

Comment:
@luispollo I'd love for you to help me understand this behavior. If I have the mock return a constraint status for `manual-judgement` another gets added with this state. I don't quite grok what's going on here.